src/main/java/org/seqdoop/hadoop_bam/BAMRecordReader.java

```diff
@@ -126,62 +127,74 @@ public static long getKey0(int refIdx, int alignmentStart0) {
 		if(isInitialized)
 			close();
 		isInitialized = true;
+		reachedEnd = false;
 
 		final Configuration conf = ctx.getConfiguration();
 
 		final FileVirtualSplit split = (FileVirtualSplit)spl;
 		final Path             file  = split.getPath();
 		final FileSystem       fs    = file.getFileSystem(conf);
 
-		this.stringency = SAMHeaderReader.getValidationStringency(conf);
+		ValidationStringency stringency = SAMHeaderReader.getValidationStringency(conf);
 
-		final FSDataInputStream in = fs.open(file);
+		java.nio.file.Path index = SamFiles.findIndex(NIOFileUtil.asPath(fs.makeQualified(file).toUri()));
+		Path fileIndex = index == null ? null : new Path(index.toUri());
+		SeekableStream inStream = fileIndex == null ? null : WrapSeekable.openPath(fs, fileIndex);
```

Comment:
If this were called indexStream it would be a lot clearer. Also this is another good try-with-resource candidate.